### ISSUE TITLE
API Metadata Functions

### ISSUE THREAD

**Issue by lehitoskin** [FLIF-hub/FLIF#360]
There needs to be more metadata functions available in the API. Having to decode a FLIF just to view its metadata is absurd considering the amount of time it takes just to decode a single file. Something along the lines of `flif_get_metadata_memory` or `flif_get_metadata_file` would be nice to have.

Another thing is setting metadata. AFAICT, after decoding the FLIF and setting the metadata, it becomes necessary to _re-encode_ the FLIF so it can be saved to disk. Now, this would be fine if you were already planning to encode the FLIF, but not if that's the only interaction to the image you were planning to do. Something along the lines of `flif_set_metadata_memory` or `flif_set_metadata_file` would be great.

tl;dr Decoding a FLIF just to read metadata is dumb. Decoding a file and then re-encoding it just to set metadata is plain unreasonable. It's easy to do both without ever encoding/decoding the file.

**Comment by jonsneyers** [comment]
I agree. Feel free to propose a pull request along these lines. To me, it's not really urgent since you can use ExifTool to do this kind of metadata-only manipulation of FLIF files. I mean, it would still be nice to have, but not urgent at this point – FLIF is not a mainstream format (yet), so full encode/decode (FLIF <-> uncompressed/PNG) is going to be way more commonly needed than transcode/manipulate (FLIF <-> FLIF).
